docs(web-modeler): switch to SBOM dependency list (#4338)
camunda/camunda-docs: docs/reference/dependencies.md

<TabItem value='web-modeler'>

- **Dependencies:** SBOM CycloneDX files with up-to-date lists of third party libraries used and their licenses can be requested on demand.

> - **Source code:** Source code access can be provided on demand.

Reviewer: ```suggestion
- **Source code:** Access to source code access is provided on [on-demand](dependency-request@camunda.com).
```

Author: ✅ Added an email address link to both bullet points.

<TabItem value='web-modeler'>

- **Dependencies:** SBOM CycloneDX files with up-to-date lists of third party libraries used and their licenses can be requested [on-demand](mailto:dependency-request@camunda.com).

> - **Source code:** Access to source code is provided on [on-demand](mailto:dependency-request@camunda.com).

Reviewer: ```suggestion
- **Source code:** Access to source code is provided [on-demand](mailto:dependency-request@camunda.com).
```

Author: Thanks and sorry for the oversight!
